tests/components/application-connector/test/compass-runtime-agent/testkit/applications/comparator_test.go

```diff
@@ -0,0 +1,192 @@
+package applications
+
+import (
+	"errors"
+	"github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	"github.com/kyma-project/kyma/tests/components/application-connector/test/compass-runtime-agent/testkit/applications/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func TestApplicationCrdCompare(t *testing.T) {
+
+	t.Run("should compare applications", func(t *testing.T) {
```

Comment:
I think You should add more test cases for testing different application data when actual application contains different data than expected.

Reply:
As discussed above it is not possible.